x/emissions/keeper/keeper.go

```diff
@@ -3482,3 +3582,263 @@ func (k *Keeper) GetPreviousForecasterScoreRatio(ctx context.Context, topicId To
+	var reputers []ActorId
+	rng := collections.NewPrefixedPairRange[TopicId, ActorId](topicId)
+	err := k.activeReputers.Walk(ctx, rng, func(key collections.Pair[TopicId, ActorId]) (bool, error) {
+		reputers = append(reputers, key.K2())
+		return false, nil
+	})
+	if err != nil {
+		return nil, errorsmod.Wrap(err, "error walking active reputers")
+	}
+	return reputers, nil
+}
+
+// ResetActiveReputersForTopic resets the active reputers for a topic
+func (k *Keeper) ResetActiveReputersForTopic(ctx context.Context, topicId TopicId) error {
+	reputerRange := collections.NewPrefixedPairRange[TopicId, ActorId](topicId)
+	if err := k.activeReputers.Clear(ctx, reputerRange); err != nil {
+		return errorsmod.Wrap(err, "error clearing active reputers")
+	}
+
+	return nil
+}
+
+// ResetActiveWorkersForTopic resets the active workers for a topic
+func (k *Keeper) ResetActiveWorkersForTopic(ctx context.Context, topicId TopicId) error {
+	infererRange := collections.NewPrefixedPairRange[TopicId, ActorId](topicId)
+	if err := k.activeInferers.Clear(ctx, infererRange); err != nil {
+		return errorsmod.Wrap(err, "error clearing active inferers")
+	}
+	forecasterRange := collections.NewPrefixedPairRange[TopicId, ActorId](topicId)
+	if err := k.activeForecasters.Clear(ctx, forecasterRange); err != nil {
+		return errorsmod.Wrap(err, "error clearing active forecasters")
+	}
+
+	return nil
+}
+
+// ResetReputersIndividualSubmissionsForTopic resets the reputer individual submissions for a topic
+func (k *Keeper) ResetReputersIndividualSubmissionsForTopic(ctx context.Context, topicId TopicId) error {
+	reputerRange := collections.NewPrefixedPairRange[TopicId, ActorId](topicId)
+	if err := k.lossBundles.Clear(ctx, reputerRange); err != nil {
+		return errorsmod.Wrap(err, "error clearing reputer losses")
+	}
+	return nil
+}
+
+// ResetWorkersIndividualSubmissionsForTopic resets the inferer individual submissions for a topic
+func (k *Keeper) ResetWorkersIndividualSubmissionsForTopic(ctx context.Context, topicId TopicId) error {
+	infererRange := collections.NewPrefixedPairRange[TopicId, ActorId](topicId)
+	if err := k.inferences.Clear(ctx, infererRange); err != nil {
+		return errorsmod.Wrap(err, "error clearing inferences")
+	}
+	forecasterRange := collections.NewPrefixedPairRange[TopicId, ActorId](topicId)
+	if err := k.forecasts.Clear(ctx, forecasterRange); err != nil {
+		return errorsmod.Wrap(err, "error clearing forecasts")
+	}
+
+	return nil
+}
+
+// SetLowestReputerScoreEma sets the lowest reputer score EMA for a topic
+func (k *Keeper) SetLowestReputerScoreEma(ctx context.Context, topicId TopicId, lowestScore types.Score) error {
+	if err := types.ValidateTopicId(topicId); err != nil {
+		return errorsmod.Wrap(err, "invalid topic id")
+	}
+	if err := types.ValidateBech32(lowestScore.Address); err != nil {
+		return errorsmod.Wrap(err, "invalid address")
+	}
+	return k.lowestReputerScoreEma.Set(ctx, topicId, lowestScore)
```

Comment:
Should fire off an event imo, and for lowest inferer and forecaster score EMA setting as well

Reply:
Should we? We already do on the reward distribution. We can make a different events for this, but I'm not sure if it will gonna have any use, at least for now. Wondering what @xmariachi thinks about this...

Reply:
Tough call.
I guess the intended usage is to have, while an epoch runs, a reading of the current lowest score, for the actors to have a more informed decision to send or not.

Once the window is closed that can be taken from the lowest score of the active inferers, however these are published when ground_truth_lag happens. (This includes the case for analytics - they can be taken from the indexer).

But this may be a very big number of events per topic/epoch (I'd say typically hundreds but can be thousands on testnet at least, although we should not optimize for testnet), and I'm not sold on the usage of this, so I'd lean towards not adding it for now, all things considered. We can emit this on-demand.